witness/src/Dscp/DB/CanProvideDB/Class.hs

```diff
@@ -4,14 +4,34 @@
 module Dscp.DB.CanProvideDB.Class where
 
 import Codec.Serialise
+import UnliftIO (MonadUnliftIO)
 
-type CanUseDb k v m = (Serialise k, Serialise v, MonadIO m, MonadThrow m)
-type CanDeleteKeys k m = (Serialise k, MonadIO m, MonadThrow m)
+type CanUseDb k v m = (Serialise k, Serialise v, MonadIO m, MonadThrow m, MonadUnliftIO m)
+
+data Batch k v
+    = Put k v
+    | Delete k
 
 data Plugin = Plugin
-    { pGet    :: forall k v m. CanUseDb      k v m =>   k      -> m (Maybe v)
-    , pPut    :: forall k v m. CanUseDb      k v m => [(k, v)] -> m ()
-    , pDelete :: forall k   m. CanDeleteKeys k   m =>  [k]     -> m ()
+    { pGet
+        :: forall   k v m
+        .  CanUseDb k v m
+        => k
+        -> m (Maybe v)
+
+    , pWrite
+        :: forall   k v m
+        .  CanUseDb k v m
+        => [Batch k v]
+        -> m ()
+
+    , pIterate
+        :: forall   k v m b
+        .  CanUseDb k v m
+        => ByteString
+        -> b
+        -> (b -> (k, v) -> b)
+        -> m b
```

Comment:
Could perfectly be written in a compact way used previously.

Reply:
Will fix